app/controllers/SurveyController.java

```diff
@@ -190,7 +190,7 @@ public Result createSurvey() throws Exception {
         UserSession session = getAuthenticatedResearcherSession();
         StudyIdentifier studyId = session.getStudyIdentifier();
         
-        Survey survey = DynamoSurvey.fromJson(requestToJSON(request()));
+        Survey survey = parseJson(request(), Survey.class);
```

Comment:
Deep in the internals of DynamoSurvey.fromJson(), it uses SurveyElementFactory.fromJson(), and this creates the correct SurveyElement subtype based on the type attribute of the JSON in the SurveyElement record. Without this, you won't have type-safe survey screens vs. questions, and these have different data. I think we need to maintain this factory method.